process_server/run_COMBINED.py

```diff
@@ -0,0 +1,579 @@
+                printlog("Setting parse error flag...",output_file=processfile,end='')
+                pflag = set_pflag_loc("parse_error")
+                if pflag == None: 
+                    printlog("Error setting flags, abort",processfile=processfile)
+                    break
+                printlog("Done, continue",output_file=processfile)
+                #continue
+            if type(exc) == ValueError:
+                printlog("Invalid data size: " + str(exc),output_file=processfile)
+                printlog("Setting datasize flag...",output_file=processfile,end='')
+                pflag = set_pflag_loc("datasize_error")
+                if pflag == None:
+                    printlog("Error setting flags, abort",processfile=processfile)
+                    break
+                printlog("Done, continue",output_file=processfile)
+                #continue
+            else:
+                clientSocket.close()
+                raise exc 
+        
+        successmsg = bytes(success + str(pflag) + '\n','utf-8')
+        printlog("Sending response...",output_file=processfile,end='')
+        while (totalbytessend < len(successmsg)):
+            totalbytessend += clientSocket.send(successmsg)
+        printlog("Done! Total bytes sent:" + str(totalbytessend) + "/" + str(len(successmsg)),output_file=processfile)
+        if pflag != 0:
+            continue
+        printlog("Data: " + str(arrData),output_file=processfile)
+
+        #if object corresponding to the image is in list
+        idx,openidx = find_id(img_id_isot,fullimg_array)
+        printlog("FIND_ID: " + str(idx) + ", " + str(openidx),output_file=processfile)#if it's not in the list, but there's an open spot, add it
+        if idx == -1 and openidx != -1:
+            #need to create new object
+            fullimg_array[openidx] = fullimg(img_id_isot,img_id_mjd,shape=tuple(np.concatenate([shape,[16]])))
+            idx = openidx
+        elif idx == -1 and openidx == -1: # shouldn't reach this case often, but if we don't have space for a new object, busy wait
+            while openidx == -1: 
+                printlog("Process server image array full, waiting for opening...",output_file=processfile,end='')
+                idx,openidx = find_id(img_id_isot,fullimg_array)
+        #otherwise, just add to the image at idx
+            	
+        #add image and update flags
+        fullimg_array[idx].add_corr_img(arrData,corr_node,args.testh23)
+        #if the image is complete, start the search
+        printlog("corrstatus:",output_file=processfile,end='')
+        printlog(fullimg_array[idx].corrstatus,output_file=processfile)
+        if fullimg_array[idx].is_full():
+            #submit a search task to the process pool
+            future = executor.submit(search_task,fullimg_array[idx],args.SNRthresh,args.subimgpix,args.model_weights,args.verbose,args.usefft,args.cluster,
```

Comment:
need to update implementation so doesnt hang with new thread

Reply:
Resolved in commit [22e0d71](https://github.com/dsa110/dsa110-nsfrb/pull/10/commits/22e0d71b5da4fb2366d38908bd35f2c16f8625e1)